src/services/offer.js

```diff
@@ -0,0 +1,65 @@
+const Offer = require("../models/Offer");
+
+class OfferService {
+    // TODO: Use typedi or similar
+    constructor() {
+
+    }
+
+    async create({
+        title,
+        publishDate,
+        endDate,
+        jobMinDuration,
+        jobMaxDuration,
+        jobStartDate,
+        description,
+        contacts,
+        isPaid,
+        vacancies,
+        jobType,
+        fields,
+        technologies,
+        isHidden,
+        owner,
+        location,
+        coordinates,
+    }) {
+        const offer = await Offer.create({
+            title,
+            publishDate,
+            endDate,
+            jobMinDuration,
+            jobMaxDuration,
+            jobStartDate,
+            description,
+            contacts,
+            isPaid,
+            vacancies,
+            jobType,
+            fields,
+            technologies,
+            isHidden,
+            owner,
+            location,
+            coordinates,
+        });
+
+        return offer;
+    }
+
+    async get() {
+        const offers = await Offer.find({
+            publishDate: {
+                $lte: new Date(Date.now()),
+            },
+            endDate: {
+                $gt: new Date(Date.now()),
+            },
+        });
+
+        return offers;
+    }
```

Comment:
Please add a `getAll` that returns the offers no matter the dates (useful for admin related tasks and not that hard to implement)

Reply:
But it would just be an unused method, or do you want me to add an endpoint for that?

If the objective is to inspect the DB's state, then the admin should just `mongo` into the URI.

Reply:
After thinking about it, I was mixing concepts. My idea was for a Company User to be able to get all his offers (current, past and future), but this is not the way, it can be implemented once we have company accounts.

Consider this proposal null

Reply:
Null it is! Resolving!